### PR TITLE
Fix submenu hiding menu in classic theme

### LINKED ISSUE #9938: [BOOM-3481] Sub menu hides menu
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-3481).

- _**Reporter:**_ kaithoc
- _**Created at:**_ Mon, 17 Jul 2017 17:47:07 +0200

<p>Hi,<br/>
I have a problem withe the menu of Prestashop classic theme.<br/>
On hover, submenu hides menu but only for the categories I created.<br/>
Thanks</p>

### PR DESCRIPTION
**Pull request by YeLnatSs**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Some submenus hides menu
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | Fixes #9938 
| How to test?  | In BO, create a category display it in main menu, add subcategory. Check in FO.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13677)

### CODE REVIEW

File: themes/classic/_dev/js/components/top-menu.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default class TopMenu extends DropDown {
       }
       if (elmId && $(e.target).data('depth') === 0) {
         $(`#${elmId} .js-sub-menu`).css({
-          top: $(`#${elmId}`).height() + $(`#${elmId}`).position().top
+          top: $(`.js-sub-menu`).parent(`#${elmId}`).height() + $(`#${elmId}`).position().top
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/top-menu.js]: Since it's an id and must be unique, why adding a `.js-sub-menu` with `parent` change? :thinking:

Comment (YeLnatSs) [reply themes/classic/_dev/js/components/top-menu.js]: The previous code with "$(`#${elmId}`)" only  wasn't working

Comment (PierreRambaud) [reply themes/classic/_dev/js/components/top-menu.js]: I think the problem from somewhere else.
Selecting a parent with this id is the same as `$(#${elmId})`

Comment (YeLnatSs) [reply themes/classic/_dev/js/components/top-menu.js]: I know. Calling only "$('#{elmId}')" set element to height:0 using the other code sets this one to height:29px like its siblings.

Comment (PierreRambaud) [reply themes/classic/_dev/js/components/top-menu.js]: So, we need to fix the problem where it start, not adding a weird patch :/

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @YeLnatSs

**Pull request merged**